streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -318,6 +320,16 @@ public void foreach(ForeachAction<? super K, ? super V> action) {
     }
 
     @Override
+    public KStream<K, V> peek(ForeachAction<? super K, ? super V> action) {
```

Comment:
please add `final` for parameter and wherever else possible (i.e., also for vars within this method, and all other classes you add in this PR)